app/src/main/java/com/concordium/wallet/data/walletconnect/ParamsDeserializer.kt

```diff
@@ -0,0 +1,66 @@
+package com.concordium.wallet.data.walletconnect
+
+import com.google.gson.JsonDeserializationContext
+import com.google.gson.JsonDeserializer
+import com.google.gson.JsonElement
+import com.google.gson.JsonParseException
+import java.lang.reflect.Type
+
+class ParamsDeserializer : JsonDeserializer<Params?> {
+    @Throws(JsonParseException::class)
+    override fun deserialize(
+        jElement: JsonElement,
+        typeOfT: Type?,
+        context: JsonDeserializationContext
+    ): Params {
+        val jObject = jElement.asJsonObject
+
+        return Params(
+            type = jObject["type"]?.asString,
```

Comment:
These `asString` calls can produce `ClassCastException` and `IllegalStateException`. These are caught below but not here. Why is that?

Reply:
I would prefer the try/catch wrapping to be in here instead to keep it more consistent. And then replace the existing try/catch with if null.

Reply:
The try/catch gives us greater flexibility later on I we  want to handle the errors and show them to the user.

Reply:
I would agree if not for the fact that the same logic is used for the schema object, meaning that it's now in two separate places. But it's up to you.